src/mono/mono/metadata/assembly.c

```diff
@@ -1475,19 +1475,19 @@ struct AssemblySearchHook {
 static AssemblySearchHook *assembly_search_hook = NULL;
 
 static MonoAssembly*
-mono_assembly_invoke_search_hook_internal (MonoAssemblyLoadContext *alc, MonoAssembly *requesting, MonoAssemblyName *aname, gboolean refonly, gboolean postload)
+mono_assembly_invoke_search_hook_internal (MonoAssemblyLoadContext *alc, MonoAssembly *requesting, MonoAssemblyName *aname, gboolean postload)
 {
 	AssemblySearchHook *hook;
 
 	for (hook = assembly_search_hook; hook; hook = hook->next) {
-		if ((hook->refonly == refonly) && (hook->postload == postload)) {
+		if ((hook->refonly == FALSE) && (hook->postload == postload)) {
```

Comment:
Can't you remove the `refonly` check entirely?

Reply:
you can still install refonly search hooks.